src/apps/aave/positions.ts

```diff
@@ -76,6 +81,30 @@ const hook: PositionsHook = {
           allowFailure: false,
         })
       : [undefined, undefined]
+
+    const [totalSupplies, lpTokenDecimals] = await Promise.all([
+      await Promise.all(
+        reservesData.map(async ({ aTokenAddress }) => {
+          return client.readContract({
+            address: aTokenAddress,
+            abi: aTokenAbi,
+            functionName: 'totalSupply',
+            args: [],
+          })
+        }),
+      ),
+      await Promise.all(
+        reservesData.map(async ({ aTokenAddress }) => {
+          return client.readContract({
+            address: aTokenAddress,
+            abi: aTokenAbi,
+            functionName: 'decimals',
+            args: [],
+          })
+        }),
+      ),
+    ])
```

Comment:
Do we need the awaits and asyncs on the `Promise.all` nested in the other `Promise.all`?

```suggestion
    const [totalSupplies, lpTokenDecimals] = await Promise.all([
     Promise.all(
        reservesData.map(({ aTokenAddress }) => {
          return client.readContract({
            address: aTokenAddress,
            abi: aTokenAbi,
            functionName: 'totalSupply',
            args: [],
          })
        }),
      ),
     Promise.all(
        reservesData.map(({ aTokenAddress }) => {
          return client.readContract({
            address: aTokenAddress,
            abi: aTokenAbi,
            functionName: 'decimals',
            args: [],
          })
        }),
      ),
    ])
```